File: egolint/tests/fixtures/lintr/valid.R
#' Format a greeting for a supplied name.
#'
#' @param name Character value containing the name to greet.
#'
#' @return A formatted greeting.
format_greeting <- function(name) {
  normalized_name <- trimws(name)

  if (!nzchar(normalized_name)) {
    return("Hello.")
  }

  sprintf("Hello, %s.", normalized_name)
}

File: egolint/tests/fixtures/prettier/valid.ts
interface GreetingOptions {
  readonly excited?: boolean;
}

export function formatGreeting(
  name: string,
  options: GreetingOptions = {},
): string {
  const normalizedName = name.trim();
  const punctuation = options.excited ? "!" : ".";

  return normalizedName === ""
    ? `Hello${punctuation}`
    : `Hello, ${normalizedName}${punctuation}`;
}
